pkg/queue/breaker_test.go

```diff
@@ -131,22 +131,19 @@ func TestBreakerCancel(t *testing.T) {
 	cancel1()
 	reqs.expectFailure(t)
 
-	// Let through a request with capacity then timeout following request
+	// Let through a request with capacity and cache a request.
 	b.UpdateConcurrency(1)
 	reqs.request()
-
-	// Exceed capacity and assert failure. This makes sure the Breaker is consistently
-	// at capacity.
 	reqs.request()
-	reqs.expectFailure(t)
 
 	// This request cannot get capacity.
 	ctx2, cancel2 := context.WithCancel(context.Background())
 	reqs.requestWithContext(ctx2)
 	cancel2()
 	reqs.expectFailure(t)
 
-	// The request that was put in earlier should succeed.
+	// The requests that were put in earlier should succeed.
+	reqs.processSuccessfully(t)
```

Comment:
We probably still need a test that overflows the queue as well?

Reply:
Done.

Actually the second canceled request fails due to overflows. But it's confusing. So I separate the failing cases.